src/protocols/bitcoin/libs/BitcoinAdapter.ts

```diff
@@ -205,9 +206,30 @@ export class BitcoinAdapter extends BaseProtocolAdapter {
     rawAccount: IAccountRaw,
     idx: number,
     globalIdx: number,
-    seed: Uint8Array,
+    seed?: Uint8Array,
   ): IAccount | null {
-    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet) {
+    const { activeNetwork } = useNetworks();
+
+    if (rawAccount.type === ACCOUNT_TYPES.privateKey && rawAccount.privateKey) {
+      const ECPair = ECPairFactory(ecc);
+      const account = ECPair.fromPrivateKey(Buffer.from(rawAccount.privateKey));
+
+      const network = networks[activeNetwork.value.type === NETWORK_TYPE_MAINNET ? 'bitcoin' : 'testnet'];
+      const { address } = payments.p2wpkh({ // p2wpkh for Native-Segwit and p2pkh for Legacy
+        pubkey: account.publicKey,
+        network,
+      });
+      return {
+        idx,
+        globalIdx,
+        secretKey: rawAccount.privateKey!,
+        ...rawAccount,
+        publicKey: account.publicKey,
+        privateKey: undefined,
+        address,
+      } as IAccount;
+    }
+    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet || !seed) {
```

Comment:
This tesrt makes no sense here. `null` should be the default return value at the end of the method.  And each other case should be handled before. For example with the use of `switch..case`.